test/unit/io_clock.js

```diff
@@ -0,0 +1,34 @@
+var test = require('tap').test;
+var Clock = require('../../src/io/clock');
+var Runtime = require('../../src/engine/runtime');
+
+test('spec', function (t) {
+    var rt = new Runtime();
+    var c = new Clock(rt);
+
+    t.type(Clock, 'function');
+    t.type(c, 'object');
+    t.type(c.projectTimer, 'function');
+    t.type(c.pause, 'function');
+    t.type(c.resume, 'function');
+    t.type(c.resetProjectTimer, 'function');
+    t.end();
+});
+
+test('cycle', function (t) {
+    var rt = new Runtime();
+    var c = new Clock(rt);
+
+    t.ok(c.projectTimer() <= 0.1);
+    setTimeout(function () {
+        c.resetProjectTimer();
+        setTimeout(function () {
+            t.ok(c.projectTimer() > 0);
+            c.pause();
+            t.ok(c.projectTimer() > 0);
+            c.resume();
+            t.ok(c.projectTimer() > 0);
+            t.end();
+        }, 100);
+    }, 100);
```

Comment:
It's not that long, but given the resolution of the timer, could these timeouts be smaller for what's being tested?
I am wondering if it would be useful to flesh the test out a bit more so that it can tell if the pause and resume methods are functioning? It would pass if the timer was paused for the entirety of the inner `setTimeout`, or if it never paused even after pause was called.  Is this the place for that test?